src/client/visualizations/heat-map/heat-map.tsx

```diff
@@ -20,162 +20,66 @@
 // tslint:disable-next-line: no-reference
 /// <reference path="../../index.d.ts" />
 
-import { TooltipProps, TooltipWithBounds, withTooltip } from "@vx/tooltip";
-import { Dataset, Datum } from "plywood";
+import memoizeOne from "memoize-one";
+import { Dataset } from "plywood";
 import * as React from "react";
-import { ConcreteSeries, SeriesDerivation } from "../../../common/models/series/concrete-series";
-import { SortDirection } from "../../../common/models/sort/sort";
-import { Split, SplitType } from "../../../common/models/split/split";
-import { VisualizationProps } from "../../../common/models/visualization-props/visualization-props";
-import { formatSegment } from "../../../common/utils/formatter/formatter";
+import { ConcreteSeries } from "../../../common/models/series/concrete-series";
 import { HEAT_MAP_MANIFEST } from "../../../common/visualization-manifests/heat-map/heat-map";
-import { MeasureBubbleContent } from "../../components/measure-bubble-content/measure-bubble-content";
-import { SegmentBubbleContent } from "../../components/segment-bubble/segment-bubble";
 import { SPLIT } from "../../config/constants";
-import {
-  fillDatasetWithMissingValues,
-  Order,
-  orderByNumberRangeDimensionDecreasing,
-  orderByNumberRangeDimensionIncreasing,
-  orderByTimeDimensionDecreasing,
-  orderByTimeDimensionIncreasing,
-  orderByValueDecreasing,
-  orderByValueIncreasing
-} from "../../utils/dataset/dataset";
-import { JSXNode } from "../../utils/dom/dom";
+import { fillDatasetWithMissingValues } from "../../utils/dataset/dataset";
 import { BaseVisualization, BaseVisualizationState } from "../base-visualization/base-visualization";
 import "./heat-map.scss";
-import { RectangleData } from "./heatmap-rectangles";
-import { LabelledHeatmap } from "./labelled-heatmap";
+import { LabelledHeatmap, TILE_SIZE } from "./labeled-heatmap";
+import scales from "./utils/scales";
 
-type HeatmapState = BaseVisualizationState & { preparedDataset: Dataset };
+interface HeatmapState extends BaseVisualizationState {
+  preparedDataset: Dataset;
+}
 
 export class HeatMap extends BaseVisualization<HeatmapState> {
   protected className = HEAT_MAP_MANIFEST.name;
 
+  getScales = memoizeOne(scales);
```

Comment:
Do you have some issues with React lifecycle or what? For what reason to memoize only the last call?

Reply:
If this function params change, there’s a lot of work to do elsewhere, and savings there would have minimal impact. That way I don’t need to handle buffer of values.